contracts/SponsorFaucet.sol

```diff
@@ -35,27 +35,21 @@ contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard {
     }
      
     /*** Dapp dev calls ***/ 
-    //apply cfx for gas & storage
-    function applyFor(address dapp) public nonReentrant whenNotPaused {
-        cpc.setSponsorForGas.value(gas_bound)(dapp, upper_bound);
-        cpc.setSponsorForCollateral.value(collateral_bound)(dapp);
-        dapps[dapp].gas_cnt.add(1);
-        dapps[dapp].collateral_cnt.add(1);
-        dapps[dapp].limit.add(gas_bound.add(collateral_bound));
-        emit applied(msg.sender, dapp, upper_bound);
-    }
-
     function applyForGas(address dapp) public nonReentrant whenNotPaused {
+        require(address(this).balance >= gas_bound, "faucet out of money");
+        require(cpc.getSponsoredBalanceForGas(dapp) < gas_bound, "sponsor value too low");
```

Comment:
Again, better to use meaningful name for sponsor contract, `cpc` is really a bad name.

BTW, it seems the error message should be "sponsored balance for gas is too high".

Reply:
cpc removed and error message fixed